core/src/test/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -312,6 +312,25 @@ public static double randomDouble() {
         return random().nextDouble();
     }
 
+    /**
+     * Returns a double value in the interval [start, end) if lowerInclusive is set to true,
+     * (start, end) otherwise.
+     *
+     * @param start lower bound of interval to draw uniformly distributed random numbers from
+     * @param end upper bound
+     * @param lowerInclusive whether or not to include lower end of the interval
+     * */
+    public static double randomDoubleBetween(double start, double end, boolean lowerInclusive) {
```

Comment:
Its used in some of the new GeoDataGenerator methods which are in turn used in the geo builder test. Some of them are from #11969 but I already picked them while continuing PR from @MaineC .